src/main.cpp

```diff
@@ -5818,8 +5828,6 @@ bool InitBlockIndex(const CChainParams& chainparams)
             CBlockIndex *pindex = AddToBlockIndex(block, chainparams.GetConsensus());
             if (!ReceivedBlockTransactions(block, state, chainparams, pindex, blockPos))
                 return error("LoadBlockIndex(): genesis block not accepted");
-            if (!ActivateBestChain(state, chainparams, &block))
-                return error("LoadBlockIndex(): genesis block cannot be activated");
```

Comment:
This is the cause of the RPC test failure. Upstream removed this call to avoid an immediate full chain activation, instead relying on the background thread. But this call was as a side-effect initialising the empty Sprout tree for the genesis block, and the RPC test that's failing is importing a spending key immediately after, which triggers `CWallet::ScanForWalletTransactions`, which ssumes we can always get the Sprout anchor for the chain tip.